new-components/targets/git-clone/Dockerfile

```diff
@@ -0,0 +1,42 @@
+# Used only to build Go binaries.
+FROM golang:1.23.3 AS builder
+
+WORKDIR /wrk
+
+# Copy only go related files.
+COPY go.mod go.sum ./
+COPY vendor ./vendor
+COPY cmd ./cmd
+COPY pkg ./pkg
+
+# Security hardening and building flags for minimal binaries.
+#
+# These CGO_CPPFLAGS help preventing overflows.
+# Add a small overhead at compile time.
+ENV CGO_CPPFLAGS="-D_FORTIFY_SOURCE=2 -fstack-protector-all"
+# Makes memory exploitation harder.
+# Add a small overhead at compile time.
+ENV GOFLAGS="-buildmode=pie"
+ENV CGO_ENABLED=0
+ENV GOOS="linux"
+ENV GOARCH="amd64"
+RUN go build -ldflags "-s -w" -trimpath -o app ./cmd/main.go
```

Comment:
can we move this into a makefile, so that it can be used both inside and outside of the container?

Reply:
yes. but I can see someone wanting to run it locally and in that case they would need to figure out how to build it themselves. having a Makefile with a build target means that we can invoke the same make target inside or outside the Dockerfile and get a binary

Reply:
Done!